### ISSUE TITLE
Validation rule for a Map that may also be null is not being generated correctly

### ISSUE THREAD

**Issue by peterhav** [firebase/bolt#123]
I have a bolt file containing the following attribute inside a type definition:

```javascript
ownedEvents:Map<KeyString, Boolean>|Null
```

The generated rules file contains a validation that does not comply with this statement:

```javascript
"ownedEvents": {
          "$key2": {
            ".validate": "$key2.length > 0 && $key2.length <= 255 && newData.isBoolean()"
          },
          ".validate": "newData.val() == null"
        },
```

The .validate of "ownedEvents" is incorrect: it does not allow creation of the list.

**Comment by mckoss** [comment]
|Null is assumed for Map's - just leave it out.  But yes, this is a bug currently.

**Comment by mckoss** [comment]
See Issue #118

**Comment by mckoss** [comment]
Fixed by #167
